src/apps/slackapp/slackapp/bolt_app.py

```diff
@@ -46,22 +48,32 @@ def hello_command(ack, body):
     ack(f"Hi, <@{user_id}>!")
 
 
-def contains_verbose(query: str) -> bool:
-    """looks for -verbose in the question and returns True or False"""
-    return "-verbose" in query.lower()
+def process_chat_history(messages: List[dict]) -> List[BaseMessage]:
+    results = []
 
+    for message in messages:
+        logger.info(message)
+        if message["type"] != "message" and message["type"] != "text":
+            continue
 
-def contains_verbosex(query: str) -> bool:
-    """looks for -verbosex in the question and returns True or False"""
-    return "-verbosex" in query.lower()
+        message_cls = AIMessage if message["user"] == self.ai_id else HumanMessage
+        # replace the at in the message with the name of the bot
+        text = message["text"].replace(f"@{self.ai_id}", f"@{self.ai_name}")
+        # added by JF
+        text = text.split("#verbose", 1)[0]  # remove everything after #verbose
+        text = text.replace("-verbose", "")  # remove -verbose if it exists
+        results.append(message_cls(content=text))
+
+    return results
 
 
 def get_response(
```

Comment:
This function is quite long. Think about breaking into smaller chunks and/or adding a docstring